src/swarm/node/simplified/NodeTemplate.d

```diff
@@ -0,0 +1,220 @@
+/*******************************************************************************
+
+    Swarm node template.
+
+    Extends swarm.node.simplified.NodeBase with the concrete legacy connection
+    handler.
+
+    TODO: this module is a replacement of the deprecated
+    swarm.node.model.Node : NodeBase. The difference is that this class melds in
+    all neo functionality. When the deprecated module is removed, this module
+    may be moved into its place.
+
+    copyright:
+        Copyright (c) 2011-2017 sociomantic labs GmbH. All rights reserved
+
+    License:
+        Boost Software License Version 1.0. See LICENSE.txt for details.
+
+*******************************************************************************/
+
+module swarm.node.simplified.NodeTemplate;
+
+import ocean.net.server.connection.IConnectionHandler;
+import swarm.node.simplified.LegacyConnectionHandlerBase;
+import swarm.node.simplified.NodeBase;
+
+/*******************************************************************************
+
+    Node base template.
+
+    Template params:
+        ConnHandler = type of connection handler (the node contains a
+            SelectListener instance which owns a pool of instances of this type)
+
+*******************************************************************************/
+
+public class NodeTemplate ( ConnHandler : LegacyConnectionHandlerBase ) : NodeBase
```

Comment:
This was the first thing I wondered. `NodeBase` is used, though: it's passed to the connection handlers, in this new framework. The other approach would be to make the connection handler templated on the type of node (i.e. an instantiation of `NodeTemplate`). I _think_ this would end up more complicated.